spack.yaml

```
@@ -2,37 +2,45 @@
 #
 # It describes a set of packages to be installed, along with
 # configuration settings.
-# This is testing of the dev-121-multi-target-workflows branch!!!
 spack:
-  definitions:
-    - ROOT_PACKAGE:
-        - access-test@git.2025.03.1
-    # Specific case for Gadi - should be run
-    - when: env['DEPLOYMENT_TARGET'] == 'gadi'
-      compiler_targets:
-        - intel@2021.10.0 target=x86_64
+  ## This section is what would be done for multi-HPC builds - see the commented reference to
+  ## $ROOT_SPEC in spack.specs below.
+  # definitions:
+  #   - ROOT_PACKAGE:
+  #       - access-test@git.2025.03.1
+  #
+  #   # Specific case for Gadi - should be run. Note the check for the environment variable first!
+  #   - when: '"DEPLOYMENT_TARGET" in env and env["DEPLOYMENT_TARGET"] == "gadi"'
+  #     compiler_targets:
+  #       - intel@2021.10.0 target=x86_64
+  #
+  #   # Example default case, should only be used outside of the CI system.
+  #   - when: '"DEPLOYMENT_TARGET" not in env'
+  #     compiler_targets:
+  #       - intel@2021.10.0 target=x86_64
 
-    # Default case, shouldn't be used
-    - when: '"DEPLOYMENT_TARGET" not in env'
-      compiler_targets:
-        - gcc@11.2.0 target=x86_64
-
-    - ROOT_SPEC:
-      - matrix:
-          - [$ROOT_PACKAGE]
-          - [$%compiler_targets]
+  #   - ROOT_SPEC:
+  #     - matrix:
+  #         - [$ROOT_PACKAGE]
+  #         - [$%compiler_targets]
   specs:
-    - $ROOT_SPEC
+    # - $ROOT_SPEC
+    - access-test@git.2025.04.000
```

```suggestion
    - access-test@git.2025.04.000=latest
```
See the access-test SBR. We are going to explicitly have a RHS because Spack upstream are going to implicitly insert `develop` if there is no RHS.

---

I thought that were exempting the root spec from that constraint, for simplicity.

---

Why would we do that?

---

You're right, we haven't been adding a RHS for the root specs.